taskvine/src/manager/vine_manager.c

```diff
@@ -2609,6 +2608,18 @@ static void reap_task_from_worker(struct vine_manager *q, struct vine_worker_inf
 
 	t->worker = 0;
 
+	switch(t->state)
+	{
+		case VINE_TASK_RUNNING:
+			list_remove(q->running_list, t);
+			break;
+		case VINE_TASK_WAITING_RETRIEVAL:
+			list_remove(q->waiting_retrieval_list, t);
+			break;
+		default:
```

Comment:
Hmm, it's odd that only two cases are handled in this switch.

If it is possible that the other cases may occur, then handle them.

If it's not possible for other cases to occur here (based on where it was called) then add an assert or some error checking to ensure that we maintain the invariant.